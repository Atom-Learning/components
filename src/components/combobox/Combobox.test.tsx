import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Combobox } from '.'

describe("Combobox component", () => {
  it('renders', async () => {
    const { container } = render(<Combobox />)

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<Combobox />)

    expect(await axe(container)).toHaveNoViolations()
  })
})