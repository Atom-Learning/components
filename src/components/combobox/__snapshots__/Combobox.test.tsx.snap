// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combobox component renders 1`] = `
.sxozp0c {
  -webkit-appearance: none;
  appearance: none;
  background-image: url(data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224px%22%20height%3D%2224px%22%20viewBox%3D%220%200%2024%2024%22%20stroke%3D%22%230f66b3%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20fill%3D%22none%22%20color%3D%22%230f66b3%22%3E%3Cpolyline%20points%3D%226%2010%2012%2016%2018%2010%22%2F%3E%3C%2Fsvg%3E);
  background-position: right var(--sx-space-3) top 50%, 0 0;
  background-repeat: no-repeat, repeat;
  background-size: var(--sx-sizes-2) auto, 100%;
  border: 1px solid var(--sx-colors-tonal500);
  border-radius: var(--sx-radii-0);
  box-shadow: none;
  box-sizing: border-box;
  color: var(--sx-colors-tonal900);
  cursor: text;
  display: block;
  font-family: var(--sx-fonts-sans);
  font-size: var(--sx-fontSizes-md);
  height: var(--sx-sizes-4);
  padding: var(--sx-space-3);
  width: 100%;
  transition: all 100ms ease-out;
}

.sxozp0c:focus-within {
  border-color: var(--sx-colors-primary900);
  box-shadow: inset 0 0 0 1px var(--sx-colors-primary900);
  outline: none;
}

.sxozp0c[disabled] {
  background-color: var(--sx-colors-tonal300);
  color: var(--sx-colors-tonal700);
  cursor: not-allowed;
}

.sxnok79 {
  list-style: none;
  margin: 0;
  padding: 0;
  -webkit-user-select: none;
  user-select: none;
}

.sxheyap {
  cursor: pointer;
  margin: 0;
  padding: var(--sx-space-2);
}

.sxheyap:hover {
  background: var(--sx-colors-tonal200);
  border-radius: var(--sx-radii-0);
}

.sxheyap[aria-selected="true"] {
  background: var(--sx-colors-alpha200);
}

.sxheyap [data-user-value] {
  font-weight: bold;
}

.sxdf19r {
  background: white;
  border: solid 1px var(--sx-colors-tonal500);
  border-radius: var(--sx-radii-0);
  box-shadow: var(--sx-shadows-0);
  box-sizing: border-box;
  font-family: var(--sx-fonts-sans);
  font-size: var(--sx-fontSizes-md);
  outline: none;
  padding: var(--sx-space-2);
  transform: translateY(var(--sx-space-2));
}

.sx7k4x5 {
  --reach-combobox: 1;
}

<div>
  <div
    class="sx7k4x5"
    data-reach-combobox=""
    data-state="idle"
  >
    <input
      aria-autocomplete="both"
      aria-controls="listbox--1"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Favourite fruit"
      class="sxozp0c"
      data-reach-combobox-input=""
      data-state="idle"
      id="fruit"
      role="combobox"
      value=""
    />
  </div>
</div>
`;
